src/rpc/blockchain.cpp

```diff
@@ -1868,23 +1868,26 @@ static UniValue getchaintxstats(const JSONRPCRequest& request)
             }.Check(request);
 
     const CBlockIndex* pindex;
-    int blockcount = 30 * 24 * 60 * 60 / Params().GetConsensus().pos.nTargetSpacing; // By default: 1 month
+    int blockcount{};
 
-    if (request.params[1].isNull()) {
-        LOCK(cs_main);
-        pindex = ::ChainActive().Tip();
-    } else {
-        uint256 hash(ParseHashV(request.params[1], "blockhash"));
+    {
         LOCK(cs_main);
-        pindex = LookupBlockIndex(hash);
-        if (!pindex) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
-        }
-        if (!::ChainActive().Contains(pindex)) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Block is not in main chain");
+        blockcount = 30 * 24 * 60 * 60 / GetTargetSpacing(*pcustomcsview); // By default: 1 month
```

Comment:
this blockcount value is not used anywhere right? its overriden before any usage.

Reply:
It is used in the first conditional after this block. Not it is the arg to std::min.
```
    if (request.params[0].isNull()) {
        blockcount = std::max(0, std::min(blockcount, pindex->nHeight - 1));
    }
```

Reply:
damn, sorry. overlooked that one...